### PR TITLE
Add timeline and checkpoint checks to backup.

### PR DESCRIPTION
**Pull request by dwsteele**
Add the following checks:

1) Checkpoint is updated in pg_control after pg_start_backup(). This helps ensure that PostgreSQL and pgBackRest have a consistent view of the storage and that PGDATA paths match.
2) Timeline of backup start WAL file matches pg_control. Hard to see how this one could get hit, but we have the power...
3) Standby is on the same timeline as the primary. If not, this standby is not following the primary.
4) Last standby checkpoint is not greater than the backup checkpoint. If so, this standby is not following the primary.

This also requires some additional plumbing to read/write timeline/checkpoint from pg_control and parse timelines from WAL filenames. There were some changes in the backup tests caused by the fact that pg_control now has different contents for each backup.

The check to ensure that the required checkpoint was reached on the standby should also be updated to use pg_control (it currently uses pg_control_checkpoint()), but that requires non-trivial changes to the test harness and will need to wait.

### CODE REVIEW

**Review by pgstef — APPROVED** [review]
Hi,

Except the little question about the error message, I don't have any specific remark about the code itself.

I agree with the interest of this PR, so +1 :-)

I ran my usual test suite against this branch and didn't found any regression. I tried to play a little bit with it and successfully encountered the error `ERROR: [058]: standby is on timeline 3 but expected 2` on a 3-node cluster with cascaded replication when the 1st standby is promoted. So, that's a nice safety-net I think.
Not sure how to simulate any problem regarding pg_control not being updated by the checkpoint requested by pg_start_backup though...

Anyway, this PR looks good to me :-)

File: src/db/db.c [hunk 1]
```diff
@@ -518,20 +545,35 @@ dbList(Db *this)
...
+        // Check that the timeline matches the primary
+        if (dbPgControl(this).timeline != targetTimeline)
+            THROW_FMT(DbMismatchError, "standby is on timeline %u but expected %u", dbPgControl(this).timeline, targetTimeline);
+
+        // Standby checkpoint before the backup started must be <= the target LSN. If not, it indicates that the standby was ahead
+        // of the primary and cannot be following it.
+        if (dbPgControl(this).checkpoint > pgLsnFromStr(targetLsn))
+        {
+            THROW_FMT(
+                DbMismatchError, "standby checkpoint '%s' is ahead of ahead of target '%s'",
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply src/db/db.c]: Oops! Fixed at 222033d0d9bf3.

**Comment by dwsteele** [comment]
> Not sure how to simulate any problem regarding pg_control not being updated by the checkpoint requested by pg_start_backup though..

Hopefully this is not possible, but it checks that our view of the data is consistent with Postgres, i.e. if Postgres writes something we should be able to see it immediately.

File: src/db/db.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (pgstef) [reply src/db/db.c]: :thumbsup:

**Review by jreidthompson — APPROVED** [review]
This looks good to me; nothing has stood out.

**Pull request merged**